packages/client/tests/functional/referentialActions-setDefault/tests_1-to-n.ts

```diff
@@ -0,0 +1,250 @@
+import { Providers } from '../_utils/providers'
+import { ConditionalError } from '../_utils/relationMode/conditionalError'
+import testMatrix from './_matrix'
+
+/* eslint-disable @typescript-eslint/no-unused-vars, jest/no-identical-title */
+
+// @ts-ignore this is just for type checks
+declare let prisma: import('@prisma/client').PrismaClient
+
+// @ts-ignore
+const describeIf = (condition: boolean) => (condition ? describe : describe.skip)
+
+/**
+ * 1:n relation
+ */
+testMatrix.setupTestSuite(
+  (suiteConfig, suiteMeta) => {
+    const conditionalError = ConditionalError.new()
+      .with('provider', suiteConfig.provider)
+      .with('providerFlavor', suiteConfig.providerFlavor)
+      // @ts-ignore
+      .with('relationMode', 'foreignKeys' as const)
+
+    describe('1:n mandatory (explicit)', () => {
+      const userModel = 'userOneToMany'
+      const postModel = 'postOneToMany'
+      const { defaultUserId } = suiteConfig
+
+      // - create user 1 with one post having id 1
+      // - create user 3 with no posts
+      async function createTemplate() {
+        // creating user id=1
+        await prisma[userModel].create({
+          data: { id: 1 },
+        })
+
+        // creating user id=${defaultUserId}
+        await prisma[userModel].create({
+          data: { id: defaultUserId },
+        })
+
+        // creating post id=1, userId=1
+        await prisma[postModel].create({
+          data: {
+            id: 1,
+            userId: 1,
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        await prisma.$transaction([prisma[postModel].deleteMany(), prisma[userModel].deleteMany()])
+      })
+
+      describe('[create]', () => {
+        test('[create] creating a table with SetDefault is accepted', async () => {
```

Comment:
How does this test "creating a table"? Doesn't this test a query, and if the table could not have been created it would have failed long before?

Reply:
> Doesn't this test a query, and if the table could not have been created it would have failed long before?

That is correct, this tests an `INSERT` query that happens after Prisma runs the `CREATE TABLE` statements under the hood. However, this test is only run on providers which accept the `SET DEFAULT` definitions on the DDL (e.g., MySQL 8+). The scenario you're describing is relevant for MySQL 5.6+ and old versions of MariaDB only, neither of which are tested on the CI